### PR TITLE
union transformer

### PR DESCRIPTION
**Pull request by mrmasterplan**
This transformer is so simple that it does not need a unittest

### CODE REVIEW

File: src/atc/transformers/union_transformer.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    """Returns the union of all input dataframes."""
+
+    def process(self, df: DataFrame) -> DataFrame:
+        return df
+
+    def process_many(self, datasets: dataset_group) -> DataFrame:
+        dfs = list(datasets.values())
+        df = dfs[0]
+        for other in dfs[1:]:
+            df = df.union(other)
```
Comment (LauJohansson) [inline src/atc/transformers/union_transformer.py:17]: Please keep in mind the difference between union and unionByName. https://spark.apache.org/docs/3.1.1/api/python/reference/api/pyspark.sql.DataFrame.unionByName.html

Comment (LauJohansson) [reply src/atc/transformers/union_transformer.py:17]: also remember that if the dataframes has different columns, you can set allowMissingColumns=True. Again, see documentation.

File: src/atc/transformers/union_transformer.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class UnionTransformer(Transformer):
+    """Returns the union of all input dataframes."""
+
+    def process(self, df: DataFrame) -> DataFrame:
+        return df
+
+    def process_many(self, datasets: dataset_group) -> DataFrame:
+        dfs = list(datasets.values())
+        df = dfs[0]
+        for other in dfs[1:]:
```
Comment (LauJohansson) [inline src/atc/transformers/union_transformer.py:16]: what happens if the datasets only contain one dataset?

File: src/atc/transformers/union_transformer.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from atc.etl.types import dataset_group
+
+
+class UnionTransformer(Transformer):
+    """Returns the union of all input dataframes."""
+
+    def process(self, df: DataFrame) -> DataFrame:
+        return df
+
+    def process_many(self, datasets: dataset_group) -> DataFrame:
```
Comment (LauJohansson) [inline src/atc/transformers/union_transformer.py:13]: what if something earlier in the ETL has made one of the dataset = NONE. Then it cannot union. Consider something like this:
```
for df in datasets:
        if df.values() is None:
            raise NoTableException(f"The {df.key} table dataframe in the list is None.")
```

**Comment by LauJohansson** [comment]
> This transformer is so simple that it does not need a unittest

Perhaps it does need one?

What do you expect if there is only one dataframe?
What do you expect if the dataframes have different schemas?
What if a dataframe is NONE?

**Review by LauJohansson — CHANGES_REQUESTED** [review]

**Comment by mrmasterplan** [comment]
> > This transformer is so simple that it does not need a unittest
> 
> Perhaps it does need one?
> 
> What do you expect if there is only one dataframe? What do you expect if the dataframes have different schemas? What if a dataframe is NONE?

this is basically a wrapper to the union function. The answer to all those questions is basically whatever the union function does.

**Review by LauJohansson — APPROVED** [review]

**Pull request merged**